### ISSUE TITLE
ES 6.x

### ISSUE THREAD

**Issue by Beenhakker** [barseghyanartur/django-elasticsearch-dsl-drf#23]
Hi,

Just curious: when will there be support for ES 6.x? I have already based my latest application on it.

Kind regards,

Bas

**Comment by barseghyanartur** [comment]
@Beenhakker:

Not sure about that. I have based this package on ``django-elasticsearch-dsl``, which is a tiny layer around the high-level ``elasticsearch-dsl``. Still things can break (probably some deprecated imports). I suggest you make an issue at ``django-elasticsearch-dsl`` (or better, come up with a pull request). 

Once ``django-elasticsearch-dsl`` is compatible with 6.x, I'll make sure this package is also compatible.

**Comment by hrdymchl** [comment]
@barseghyanartur thanks for maintaining this. It looks like `django-elasticsearch-dsl` added support for ES 6 today. Any chance we can make this 6.x compatible too?

**Comment by barseghyanartur** [comment]
@hrdymchl:

Sure. I'll do it ASAP. Thanks for the update.

**Comment by mnzaki** [comment]
Hey @barseghyanartur what's the status on this?
What kind of help is needed?

**Comment by barseghyanartur** [comment]
@mnzaki:

Well, if you could implement Elasticsearch 6.x suppot, it would be great. I didn't really read into it, but I think the main problem is that some fields have been removed. 

 I want to keep support for Elasticsearch 2.x. That's essential.

We can declare them in the ``compat`` module (based on the ``elasticsearch`` Python package version installed).

Would you want to work on that?

**Comment by barseghyanartur** [comment]
@mnzaki:

Thanks!

It's really strange that functional suggest didn't work, since it's based on the natural Elastic queries, just mimicing the suggest functionality.

Please, keep posting your findings.

**Comment by mnzaki** [comment]
Ah my bad, just suggest! I haven't actually tested functional suggest thoroughly, because I wrote my own custom end points.

**Comment by barseghyanartur** [comment]
Ah. Ok. Btw, you're advised to work with filter backends, instead of custom endpoints.

**Comment by barseghyanartur** [comment]
@mnzaki:

I tried to run tests with Elastic 6.x and got a lot of errors.

```
5 failed, 27 passed, 4 skipped, 1 pytest-warnings, 79 error.
```

**Comment by barseghyanartur** [comment]
Suggest functionality doesn't seem to work for me. An [issue opened](https://github.com/elastic/elasticsearch-dsl-py/issues/935) in elasticsearch-dsl-py.

**Comment by barseghyanartur** [comment]
**Update**

After [some work](https://github.com/barseghyanartur/django-elasticsearch-dsl-drf/tree/elasticsearch-6.x):

```
4 failed, 101 passed, 10 skipped, 1 pytest-warnings in 121.63 seconds
```

Only suggesters don't work 100%.

```
src/django_elasticsearch_dsl_drf/tests/test_faceted_search.py .
src/django_elasticsearch_dsl_drf/tests/test_filtering_common.py ........................ss.
src/django_elasticsearch_dsl_drf/tests/test_filtering_geo_spatial.py .......
src/django_elasticsearch_dsl_drf/tests/test_filtering_nested.py ..............ss
src/django_elasticsearch_dsl_drf/tests/test_filtering_post_filter.py ........................ss.
src/django_elasticsearch_dsl_drf/tests/test_functional_suggesters.py ..
src/django_elasticsearch_dsl_drf/tests/test_helpers.py .
src/django_elasticsearch_dsl_drf/tests/test_highlight.py .
src/django_elasticsearch_dsl_drf/tests/test_ordering_common.py ...........ss
src/django_elasticsearch_dsl_drf/tests/test_ordering_geo_spatial.py .
src/django_elasticsearch_dsl_drf/tests/test_pagination.py .
src/django_elasticsearch_dsl_drf/tests/test_search.py ss..
src/django_elasticsearch_dsl_drf/tests/test_serializers.py .....
src/django_elasticsearch_dsl_drf/tests/test_suggesters.py FF.FF
src/django_elasticsearch_dsl_drf/tests/test_views.py ..
src/django_elasticsearch_dsl_drf/tests/test_wrappers.py ..
```

**Comment by barseghyanartur** [comment]
@mnzaki:

In case you want to help fixing the suggesters, see previous post.

**Comment by barseghyanartur** [comment]
Update: It works. I'm releasing version 0.10 today.

**Comment by barseghyanartur** [comment]
Released in 0.10.

**Comment by mnzaki** [comment]
Nice work, thanks @barseghyanartur!
